### PR TITLE
Pytorch example

### LINKED ISSUE #9: Example tensorflow/pytorch script
It would be nice to have an MNIST-type example for both pytorch and tensorflow.
This would include the python script as well as the slurm file.

These could potentially be expanded a little further with the use for example of multiple GPUs.

### PR DESCRIPTION
**Pull request by mdiazmel**
This PR adds a Pytorch example ready to use in Jean Zay (see #9).

It uses the MNIST dataset. (~~currently asking to JZ admins to put it into the $DSDIR~~)

Fix #9.

### CODE REVIEW

**Comment by mdiazmel** [comment]
I just updated this PR. I works now with the MNIST dataset stored in the dataset folder ($DSDIR) of JZ. 

cc @lesteve

File: examples/pytorch/pytorch_example_script_multigpu.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#SBATCH --error=pytorch_mnist%j.err  # error file name
+#SBATCH --array=1-10
+
+
+cd $WORK/jean-zay-doc/examples/pytorch
+
+module purge
+module load pytorch-gpu/py3/1.4.0 
+GAMMA_STEP=('0.1' '0.2' '0.3' '0.4' '0.5' '0.6' '0.7' '0.8' '0.9' '1.0') 
+srun python ./mnist_example.py --gamma ${GAMMA_STEP[$SLURM_ARRAY_TASK_ID]} &
```
Comment (lesteve) [inline examples/pytorch/pytorch_example_script_multigpu.sh]: Nice it is the first time I see a SLURM job array in real-life! As you probably know I would have done it with `dask-jobqueue` but job arrays is definitely something useful.

Just out of curiosity:
- do you need `srun` if you are running one single task, i.e. could you use:
```
python ./mnist-example.py ...
```
- the `&` + `wait`, is it a common thing (I saw it was used in the Tensorflow examples as well)? Do you need it here since you are running a single task?

File: examples/pytorch/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Pytorch example script
+
+To run this script you will need to clone the jean-zay repo in your `$WORK`
+dir, then:
+```
+cd $WORK &&\
+git clone https://github.com/jean-zay-users/jean-zay-doc.git
+```
+
+After that, you can just launch the batch job (single GPU) via:
```
Comment (lesteve) [inline examples/pytorch/README.md]: Minor detail, is something I learned from Software Carpentry is to avoid "just". It is not useful and can be detrimental (i.e. demotivating) if the person does not understand what you mean.

File: README.md [hunk 3]
```diff
@@ -52,7 +52,6 @@ Doc: http://www.idris.fr/eng/jean-zay/
 
 Doc in French (more accurate sometimes): http://www.idris.fr/jean-zay/
 
-Email for Jean Zay user support: [assist@idris.fr](mailto:assist@idris.fr).
```
Comment (lesteve) [inline README.md]: Probably not intended I reverted back the change.

File: examples/pytorch/README.md (hunk 2 above)
Comment (lesteve) [reply examples/pytorch/README.md]: Actually re-reading this, maybe you want to mention that it launches a job array (i.e. the equivalent of 10 jobs with different learning rate parameters)

**Comment by lesteve** [comment]
Nice github tip (in case you don't know it) if you put `Fix #9` in the PR description it will close the associated issue. I added this for you.

For more details, look at [this](https://github.com/blog/1506-closing-issues-via-pull-requests).

File: examples/pytorch/pytorch_example_script_multigpu.sh (hunk 1 above)
Comment (zaccharieramzi) [reply examples/pytorch/pytorch_example_script_multigpu.sh]: Well the `&` + `wait` is a relic because I used it when I had multiple `srun`s. With the array or even with the conf file it's not needed.

I don't know about `srun`.

Comment (mdiazmel) [reply examples/pytorch/pytorch_example_script_multigpu.sh]: I copied the script from the tr example (trying to keep some coherence between the examples) but in fact I think that the `&`, `wait` and `srun` are not necessary for this kind of script. I will remove them before merging.

File: examples/pytorch/README.md (hunk 2 above)
Comment (mdiazmel) [reply examples/pytorch/README.md]: I edited this text.

**Comment by lesteve** [comment]
Great thanks a lot for this PR!
